src/kartograf/atom_mapper.py

```diff
@@ -855,6 +854,38 @@ def suggest_mapping_from_rdmols(
 
         return mapping
 
+    def _split_protein_components_molecules(self, protein: ProteinComponent) -> list[ProteinComponent]:
+        """
+        Aims at splitting a protein component into different protein components based on the
+        connectivity of the molecules that compose it. Useful for mapping multimer components
+        or proteins with structural waters or similarly.
+
+        This returns a ``ProteinComponent`` object with the name attribute indicating the starting
+        index in the original component.
+        """
+        from rdkit.Chem.rdmolops import GetMolFrags
+        rdmol = protein.to_rdkit()
+        fragments_indices = GetMolFrags(rdmol)
```

Comment:
Would it be easier to let rdkit handle the splitting of the molecule? [GetMolFrags](https://www.rdkit.org/docs/source/rdkit.Chem.rdmolops.html#rdkit.Chem.rdmolops.GetMolFrags) can return rdkit molecules and sanitize them.

Reply:
It turns out it is also much faster to let rdkit split the molecule for you in local testing the `test_split_multimeric_component` goes from 19s to 1!